hdf5.spec

```
@@ -142,8 +143,8 @@ HDF5 static libraries.
 Summary: HDF5 mpich libraries
 BuildRequires: mpich-devel
 Provides: %{name}-mpich2 = %{version}-%{release}
-Obsoletes: %{name}-mpich2 < 1.8.11-4
-Provides: %{name}-mpich2-cart-%{cart_major}-daos-%{daos_major}
+Provides: %{name}-mpich2-daos-%{daos_major}
+Obsoletes: %{name} < %{version}-%{release}
```

It does not seem correct that the `hdf5-devel` package obsoletes the `hdf5` package:
```suggestion
Obsoletes: %{name}-devel < %{version}-%{release}
```
It seems to be causing this error:
```
[2020-07-28T06:53:30.271Z] --> Processing Dependency: libhdf5.so.103()(64bit) for package: MACSio-1.1-1.el7.x86_64
[2020-07-28T06:53:30.271Z] Package hdf5 is obsoleted by hdf5-mpich, but obsoleting package does not provide for requirements
```
in https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-3136/runs/8/nodes/397/steps/593/log/?start=0